### ISSUE TITLE
Unable to access Pynbody particle data

### ISSUE THREAD

**Issue by Morkney** [pynbody/tangos#59]
Hello,

I have just recently updated Tangos and am now getting an error when writing properties with 'tangos write'. It seems to occur whenever there is an attempt to reference the halo particle data, however there is no such error if halo.properties['something'] is called. This error occurs independently of any arguments used alongside 'tangos write'.

The traceback looks like this:

```
Traceback (most recent call last):
  File "/scratch/mo00101/anaconda2/lib/python2.7/site-packages/tangos/tools/property_writer.py", line 382, in _get_property_value
    result = property_calculator.calculate(snapshot_data, db_data)
  File "/scratch/mo00101/anaconda2/lib/python2.7/site-packages/tangos/properties/pynbody/centring.py", line 61, in new_fn
    with _recenter(halo.ancestor, existing_properties['shrink_center']):
  File "/scratch/mo00101/anaconda2/lib/python2.7/site-packages/pynbody/snapshot/__init__.py", line 583, in ancestor
    return self.base.ancestor
  File "/scratch/mo00101/anaconda2/lib/python2.7/site-packages/pynbody/snapshot/__init__.py", line 583, in ancestor
    return self.base.ancestor
  File "/scratch/mo00101/anaconda2/lib/python2.7/site-packages/pynbody/snapshot/__init__.py", line 583, in ancestor
    return self.base.ancestor

<repeated many times>

  File "/scratch/mo00101/anaconda2/lib/python2.7/site-packages/pynbody/snapshot/__init__.py", line 583, in ancestor
    return self.base.ancestor
  File "/scratch/mo00101/anaconda2/lib/python2.7/site-packages/pynbody/snapshot/__init__.py", line 583, in ancestor
    return self.base.ancestor
RuntimeError: maximum recursion depth exceeded
2018-05-15 14:49:01,652 : Further errors from this calculation on this timestep will be counted but not individually reported.
2018-05-15 14:49:01,652 : Done with <TimeStep u'Halo332_DMonly/output_00067' z=0.52 t=8.50 Gyr>
2018-05-15 14:49:01,678 :             Succeeded: 0 property calculations
2018-05-15 14:49:01,678 :               Errored: 1 property calculations
2018-05-15 14:49:01,678 :   Errored during load: 0 property calculations
2018-05-15 14:49:01,678 :        Already exists: 1 property calculations
2018-05-15 14:49:01,678 : Missing pre-requisite: 0 property calculations
2018-05-15 14:49:01,678 : Attempting to commit 1 halo properties...
2018-05-15 14:49:01,678 : 1 properties were committed
2018-05-15 14:49:01,678 : CUMULATIVE RUNNING TIMES (just this node)
2018-05-15 14:49:01,678 :  ExampleHaloProperty 0.0s | 100.0%
```

All the best, -Matt

**Comment by apontzen** [comment]
Unfortunately I can't reproduce this behaviour. Please update pynbody and tangos to the latest version on git and if it persists give more information about the exact command line etc so that we might be able to reproduce the error.

**Comment by apontzen** [comment]
Thanks @morkney, I believe the zero-position/zero-velocity issue is a pynbody bug that should be fixed in branch issue-450 https://github.com/pynbody/pynbody/pull/453 and so will be fixed in pynbody release 0.45
